### PR TITLE
feat(DDI-1235): add new icon button variant

### PR DESCRIPTION
**Pull request by Radha-Alberta**
Added a new variant "destructive" in the icon button component.

### CODE REVIEW

File: libs/web-components/src/components/icon-button/IconButton.svelte [hunk 1]
```diff
@@ -102,22 +102,41 @@
...
+  button:hover:not(.destructive) {
     background-color: var(--goa-color-greyscale-100);
     border-color: var(--goa-color-greyscale-100);
     color: var(--goa-color-interactive-hover);
     outline: none;
   }
 
-  button:focus,
-  button:active {
+  button:focus:not(.destructive),
```
Comment (chrisolsen) [inline libs/web-components/src/components/icon-button/IconButton.svelte:117]: These `not`s shouldn't be required with the most specific selectors below.

File: libs/web-components/src/components/icon-button/IconButton.spec.ts [hunk 2]
```diff
@@ -16,6 +17,7 @@ describe("IconButton", () => {
       const iconButton = await baseElement.findByTestId("iconButton-test");
 
       expect(iconButton).toBeTruthy();
+      expect(iconButton.className).toContain("destructive");
```
Comment (chrisolsen) [inline libs/web-components/src/components/icon-button/IconButton.spec.ts:20]: There should be a separate test for the variants, as this test is for the margins.

File: libs/react-components/src/lib/icon-button/icon-button.spec.tsx [hunk 3]
```diff
@@ -1,14 +1,23 @@
...
+        mt="s"
+        mr="m"
+        mb="l"
+        ml="xl"
+      />
     );
     const el = container.querySelector("goa-icon-button");
 
     expect(el.getAttribute("icon")).toBe("information");
+    expect(el.getAttribute("variant")).toBe("destructive");
```
Comment (chrisolsen) [inline libs/react-components/src/lib/icon-button/icon-button.spec.tsx:20]: Could you improve these tests as well by adding a new set of tests for the variants?

File: libs/web-components/src/components/icon-button/IconButton.spec.ts (hunk 2 above)
Comment (Radha-Alberta) [reply libs/web-components/src/components/icon-button/IconButton.spec.ts]: Added new test for variants.

File: libs/react-components/src/lib/icon-button/icon-button.spec.tsx (hunk 3 above)
Comment (Radha-Alberta) [reply libs/react-components/src/lib/icon-button/icon-button.spec.tsx]: Added new test for variants.

File: libs/docs/src/components/common/IconButton.stories.mdx [hunk 4]
```diff
@@ -215,6 +216,7 @@ import { GoAIconButton, GoAButtonGroup } from "@abgov/react-components";
       code={`
         <goa-icon-button icon="close" variant="color"/>
         <goa-icon-button icon="close" variant="nocolor"/>
```
[the remark this answers is not in the document]
Comment (Radha-Alberta) [reply libs/docs/src/components/common/IconButton.stories.mdx:219]: Added example for dark variant

**Review by ArakTaiRoth — APPROVED** [review]

**Pull request merged**